app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
+  UK_POSTCODE_PATTERN = %r{
+    \A
+    ([Gg][Ii][Rr] 0[Aa]{2}) # Unusual postcodes
+    |
+    (
+      # Outward code
+      (([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))
+      \s?
+      [0-9][A-Za-z]{2} # Inward code
+    )
+    \Z
+  }x.freeze
+
+  attr_reader :query, :normalised_postcode
+  delegate :no_information?, to: :location_lookup
+
+  def initialize(query)
+    @query = query
+    # Use the uk_postcode gem to potentially transpose O/0 and I/1.
+    @normalised_postcode = UKPostcode.parse(
+      query.gsub(/[^a-z0-9 ]/i, "").strip,
+    ).to_s
+  end
+
+  def error_code
+    return "postcodeLeftBlank" if query.empty?
+    return "postcodeLeftBlankSanitized" if normalised_postcode.empty?
+    return "invalidPostcodeFormat" unless normalised_postcode.match?(UK_POSTCODE_PATTERN)
+    return "fullPostcodeNoMapitMatch" if location_lookup.postcode_not_found?
+    return "fullPostcodeNoMapitValidation" if location_lookup.invalid_postcode?
+  end
+
+  def empty?
```

Comment:
I've flipped this to be `blank_postcode?` since that seems to be a closer match to the `postcodeLeftBlank` errorCode. Does that resolve it?

I did find this a rather confusing method to try and name as it seemed rather strange that if you enter a syntactically valid postcode that Mapit doesn't know (for example SW11 9ZZ) it'll come back with an error of no postcode but you clearly put one in. I didn't want to mess with that logic in this PR though.